app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -1,15 +1,218 @@
+                    $(tooltipTriggerId).tooltip('close');
+                    $(tooltipTriggerId).attr('ttVisible','no');
+                }
+
+                $(tooltipTriggerId).unbind('mouseover');
+                $(tooltipTriggerId).attr('ttVisible','no');
+                
+                $(tooltipTriggerId).click(function() {
+                    if($(tooltipTriggerId).attr('ttVisible') == 'no') {
+                        
+                        // Open the tooltip on click
+                        $(tooltipTriggerId).tooltip('open');
+                        $(tooltipTriggerId).unbind('mouseleave');
+                        $(tooltipTriggerId).attr('ttVisible','yes');
+
+                        // Add handlers for closing the tooltip
+                        $(document).on("click", function (event) {
+                                // Target is not the tootltip or a child of the tooltip
+                                if ($(event.target).closest("#" + $(tooltipTriggerId).data("ui-tooltip-id")).length === 0 && $(event.target).closest(tooltipTriggerId).length === 0) {
+                                    closeTooltip();
+                                    $(this).unbind('click', arguments.callee);
+                                }
+                            }
+                        );
+                        $("#" + $(tooltipTriggerId).data("ui-tooltip-id") + " .ui-icon-closethick").click(closeTooltip);
+                    }
+                });
+            }
+        }
+
+        function setupPluginPlaceholders(parentDOMNode, headerSelector, contentSelector) {
+            parentDOMNode.addClass("advanced-messenger-bordered-panel");
+
+            // ====== Set the placeholders that will contain dynamically render content (that changes without page refresh) ======
+            (parentDOMNode.children('.contextual').length == 0) && (parentDOMNode = parentDOMNode.children(':first'));  
+            parentDOMNode.children('.contextual').after("<div class='read-unread-indicators-at-right' style='float: right;'><span class='read_checkbox_placeholder'></span><span class='users_initials_and_tooltip_placeholder'></span></div>");
+            
+            const expandCollapse = "<div class='expand-collapse-icon-placeholder'></div>";
+            var header = parentDOMNode.children(headerSelector);
+            var gravatar = header.find(".gravatar");
+            gravatar.length == 0 ? header.prepend(expandCollapse): gravatar.after(expandCollapse);     
+
+            parentDOMNode.children(contentSelector).html("<div class='message-content'>" + parentDOMNode.children(contentSelector).html() + "</div>");
+            parentDOMNode.children(contentSelector).append("<p class='message-preview' style='color: rgb(100, 100, 100)'></p>");
+        }
+
+        function renderPluginForMessage(id, read_by_users, preview, creationDate, messageDOMNode, headerSelector, contentSelector, updateReadByUsersUrl, users, readStatuses) {
```

Comment:
I see this is used for `message` and for `note`. Maybe a different name?

Reply:
Ok. I renamed it to ```renderPluginForMessageOrJournal```